utils/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2018-2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![warn(missing_docs)]
+
+//! Metadata utilities.
+//!
+//! This crate allows you to parse the latest metadata and use that knowledge
+//! to interact with the chain.
+
+use frame_metadata::RuntimeMetadata;
+
+pub mod error;
```

Comment:
If you rename this to `types` we avoid the weird `error::Result<Self>` stuff below. Or just `use error::{Error, Result}`.

Reply:
I actaully very much prefer to never shadow the `std::result::Result` type which is in prelude. `error::Result` always indicate that it's a custom `Result` type (hence only one generic parameter). It's a quite common pattern I think.